internal/ramses/reader.go

```diff
@@ -0,0 +1,58 @@
+package ramses
+
+import (
+	"encoding/binary"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+// Package is complete Ramses package
+type Package struct {
+	Header       Ramses
+	SecureHeader Secure
+	Payload      []byte
+}
+
+//Packages reads Ramses packages from buffer
+func Packages(buf io.Reader, packageChannel chan<- Package, errorChannel chan<- error) {
+	defer close(packageChannel)
+	defer close(errorChannel)
+	var err error
+	for {
+		header := Ramses{}
+		err = header.Read(buf)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			errorChannel <- err
+			break
+		}
+
+		if !header.Valid() {
+			errorChannel <- errors.New("Not a valid RAC-file")
```

Comment:
This isn't a very useful error, I would have liked to have a filename with it here, could we not require the streams be passed to in pair with the name/identifier cli or aws got for the file?

Reply:
it should probably also be propagated further to the packageChannel and onto the aez chan.